processing/bloom_handler.go

```diff
@@ -13,78 +12,82 @@ import (
 )
 
 var sigs = map[string]string{
-	"http-url":  "%s Possibly bad HTTP URL: ",
-	"http-host": "%s Possibly bad HTTP host: ",
-	"tls-sni":   "%s Possibly bad TLS SNI: ",
-	"dns-req":   "%s Possibly bad DNS lookup to ",
-	"dns-resp":  "%s Possibly bad DNS response for ",
+	"http-url":  "%s Possibly bad HTTP URL: %s",
+	"http-host": "%s Possibly bad HTTP host: %s",
+	"tls-sni":   "%s Possibly bad TLS SNI: %s",
+	"dns-req":   "%s Possibly bad DNS lookup to %s",
+	"dns-resp":  "%s Possibly bad DNS response for %s",
 }
 
 // MakeAlertEntryForHit returns an alert Entry as raised by an external
 // indicator match, e.g. a Bloom filter hit. The resulting alert will retain
 // the triggering event's metadata (e.g. 'dns' or 'http' objects) as well as
 // its timestamp.
 func MakeAlertEntryForHit(e types.Entry, eType string, alertPrefix string, ioc string) types.Entry {
-	var eve types.EveEvent
-	var newEve types.EveEvent
-	var err = json.Unmarshal([]byte(e.JSONLine), &eve)
+
+	var value string
+	if eType == "http-url" {
+		value = fmt.Sprintf("%s | %s | %s", e.HTTPMethod, e.HTTPHost, e.HTTPUrl)
+	} else if eType == "http-host" {
+		value = e.HTTPHost
+	} else if strings.HasPrefix(eType, "dns") {
+		value = e.DNSRRName
+	} else if eType == "tls-sni" {
+		value = e.TLSSni
+	}
+	var sig = "%s Possibly bad traffic: %s"
+	if v, ok := sigs[eType]; ok {
+		sig = v
+	}
+	newEntry := e
+	newEntry.EventType = "alert"
+	l, err := jsonparser.Set([]byte(newEntry.JSONLine), []byte("\"alert\""), "event_type")
 	if err != nil {
-		log.Warn(err, e.JSONLine)
+		log.Warning(err)
 	} else {
-		var value string
-		if eType == "http-url" {
-			value = fmt.Sprintf("%s | %s | %s", e.HTTPMethod, e.HTTPHost, e.HTTPUrl)
-		} else if eType == "http-host" {
-			value = e.HTTPHost
-		} else if strings.HasPrefix(eType, "dns") {
-			value = e.DNSRRName
-		} else if eType == "tls-sni" {
-			value = e.TLSSni
-		}
-		var sig = "%s Possibly bad traffic: "
-		if v, ok := sigs[eType]; ok {
-			sig = v
-		}
-		newEve = types.EveEvent{
-			EventType: "alert",
-			Alert: &types.AlertEvent{
-				Action:    "allowed",
-				Category:  "Potentially Bad Traffic",
-				Signature: fmt.Sprintf(sig, alertPrefix) + value,
-			},
-			FlowID:     eve.FlowID,
-			Stream:     eve.Stream,
-			InIface:    eve.InIface,
-			SrcIP:      eve.SrcIP,
-			SrcPort:    eve.SrcPort,
-			DestIP:     eve.DestIP,
-			DestPort:   eve.DestPort,
-			Proto:      eve.Proto,
-			TxID:       eve.TxID,
-			Timestamp:  eve.Timestamp,
-			PacketInfo: eve.PacketInfo,
-			HTTP:       eve.HTTP,
-			DNS:        eve.DNS,
-			TLS:        eve.TLS,
-			ExtraInfo: &types.ExtraInfo{
-				BloomIOC: ioc,
-			},
+		newEntry.JSONLine = string(l)
+	}
+	l, err = jsonparser.Set([]byte(newEntry.JSONLine), []byte("\"allowed\""), "alert", "action")
+	if err != nil {
+		log.Warning(err)
+	} else {
+		newEntry.JSONLine = string(l)
+	}
+	l, err = jsonparser.Set([]byte(newEntry.JSONLine), []byte("\"Potentially Bad Traffic\""), "alert", "category")
```

Comment:
It looks like the code is not using the `if l, err := ...; err != nil` Go thingy? It would make sense here, since `l` is then within the scope of the if statement only?

Reply:
Indeed. Changed.